### PR TITLE
Add FullScreen option to LogFrame

### PR DESCRIPTION
**Pull request by DziubanMaciej**
By default the saved framebuffers are constrained to OpenGL viewport.
It is often more comfortable to look at the entire buffer, which can
be enabled by setting FullScreen to True.

I'm leaving it OFF by default, to not disrupt well known default 
behaviour of this tool.

### CODE REVIEW

File: Src/MainLib/InterceptFrame.cpp [hunk 1]
```diff
@@ -10,6 +10,10 @@
 #include "ImageSaveManager.h"
 #include "MovieMaker.h"
 
+#ifdef GLI_BUILD_WINDOWS
```
Comment (dtrebilco) [inline Src/MainLib/InterceptFrame.cpp:13]: Is this windows.h include needed here? It should be pulled from a lot of headers that include osdefines.h which has windows.h

File: Src/MainLib/InterceptFrame.cpp [hunk 2]
```diff
@@ -903,13 +908,45 @@ bool InterceptFrame::IsBufferReadable(GLenum bufType, uint bufferCount) const
...
 
+///////////////////////////////////////////////////////////////////////////////
+//
+void InterceptFrame::GetBufferViewSize(GLint viewSize[4]) const
+{
+#ifdef GLI_BUILD_WINDOWS
+  if (this->fullscreen)
+  {
+      GLint fboId = 0;
+      GLV.glGetIntegerv(GL_READ_FRAMEBUFFER_BINDING, &fboId);
```
Comment (dtrebilco) [inline Src/MainLib/InterceptFrame.cpp:919]: Probably should check the extensionFBO flag before calling this

File: Src/MainLib/InterceptFrame.cpp [hunk 3]
```diff
@@ -903,13 +908,45 @@ bool InterceptFrame::IsBufferReadable(GLenum bufType, uint bufferCount) const
...
+                  viewSize[1] = 0;
+                  viewSize[2] = clientRect.right - clientRect.left;
+                  viewSize[3] = clientRect.bottom - clientRect.top;
+                  return;
+              }
+          }
+      }
+  }
+#endif
+
```
Comment (dtrebilco) [inline Src/MainLib/InterceptFrame.cpp:939]: If you wanted to add support for FBOs with fullscreen you could add a call to GetRenderbufferParameteriv with GL_RENDERBUFFER_WIDTH (optional)

File: Src/MainLib/InterceptFrame.cpp (hunk 1 above)
Comment (DziubanMaciej) [reply Src/MainLib/InterceptFrame.cpp:13]: Hmm, I thought it is, I will double check.

File: Src/MainLib/InterceptFrame.cpp (hunk 3 above)
Comment (DziubanMaciej) [reply Src/MainLib/InterceptFrame.cpp:939]: Hm, isn't it more complicated?

Based on [this](https://www.khronos.org/opengl/wiki/Framebuffer_Object#Attachment_Completeness),

> Notice that there is no restriction based on size. The effective size of the FBO is the intersection of all of the sizes of the bound images (ie: the smallest in each dimension). 

we would have to iterate over all attachments, check their types (we don't now whether it's a texture or a renderbuffer), query their sizes and then take the minimum.

Comment (DziubanMaciej) [reply Src/MainLib/InterceptFrame.cpp:939]: I will fiddle with this when I have more free time and get back to you later :) For now I just needed this functionality for default framebuffers.

Comment (dtrebilco) [reply Src/MainLib/InterceptFrame.cpp:939]: You don't have to do this, it was just a suggestion.

File: Src/MainLib/InterceptFrame.cpp (hunk 1 above)
Comment (DziubanMaciej) [reply Src/MainLib/InterceptFrame.cpp]: Right, it wasn't needed, I removed it.

File: Src/MainLib/InterceptFrame.cpp (hunk 3 above)
Comment (DziubanMaciej) [reply Src/MainLib/InterceptFrame.cpp:981]: I added support for FBO!

File: Src/MainLib/InterceptFrame.cpp (hunk 2 above)
Comment (DziubanMaciej) [reply Src/MainLib/InterceptFrame.cpp]: Done.

File: Src/MainLib/InterceptFrame.cpp [hunk 4]
```diff
@@ -903,13 +910,88 @@ bool InterceptFrame::IsBufferReadable(GLenum bufType, uint bufferCount) const
...
+      if (objectType == GL_TEXTURE && iglGetTexLevelParameteriv)
+      {
+        GLint mipLevel = 0;
+        iglGetFramebufferAttachmentParameteriv(GL_DRAW_FRAMEBUFFER, attachment, GL_FRAMEBUFFER_ATTACHMENT_TEXTURE_LEVEL, &mipLevel);
+
+        GLint textureName = 0;
+        iglGetFramebufferAttachmentParameteriv(GL_DRAW_FRAMEBUFFER, attachment, GL_FRAMEBUFFER_ATTACHMENT_OBJECT_NAME, &textureName);
+
+        GLint oldBinding;
+        GLV.glBindTexture(GL_TEXTURE_2D, textureName);
```
Comment (dtrebilco) [inline Src/MainLib/InterceptFrame.cpp:961]: Bind texture should happen after the old binding?

Comment (DziubanMaciej) [reply Src/MainLib/InterceptFrame.cpp:961]: Right, thanks for alertness :)

**Review by dtrebilco — COMMENTED** [review]
Initialize fullscreen boolean in ConfigData.cpp and it should be good to merge

**Pull request merged**